configuration/external_parameters.rst

```diff
@@ -80,24 +40,84 @@ You can now reference these parameters wherever you need them.
 
         <doctrine:config>
             <doctrine:dbal
-                driver="pdo_mysql"
-                dbname="symfony_project"
-                user="%database.user%"
-                password="%database.password%"
+                host="%env(DATABASE_HOST)%"
             />
         </doctrine:config>
 
     .. code-block:: php
 
         $container->loadFromExtension('doctrine', array(
             'dbal' => array(
-                'driver'   => 'pdo_mysql',
-                'dbname'   => 'symfony_project',
-                'user'     => '%database.user%',
-                'password' => '%database.password%',
+                'host' => '%env(DATABASE_HOST)%',
             )
         ));
 
+``env(...)`` parameters can have default values be set and used whenever their
+corresponding environment variables are not found:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        parameters:
+            env(DATABASE_HOST): localhost
+
+    .. code-block:: xml
+
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <parameters>
+                <parameter key="env(DATABASE_HOST)">localhost</parameter>
+            </parameters>
+         </container>
+
+    .. code-block:: php
+
+        $container->setParameter('env(DATABASE_HOST)', 'localhost');
+
+If you're using Apache, environment variables can be set e.g. using the following
+``VirtualHost`` configuration:
+
+.. code-block:: apache
+
+    <VirtualHost *:80>
+        ServerName      Symfony
+        DocumentRoot    "/path/to/symfony_2_app/web"
+        DirectoryIndex  index.php index.html
+        SetEnv          DATABASE_USER user
+        SetEnv          DATABASE_PASSWORD secret
+
+        <Directory "/path/to/symfony_2_app/web">
+            AllowOverride All
+            Allow from All
+        </Directory>
+    </VirtualHost>
+
+.. note::
+
+    All popular web servers support setting environment variables. Read their
+    documentation to know how.
+
+    Also, in order for your console to work, you must export these as shell
+    variables. On a Unix system, you can run the following:
+
+    .. code-block:: bash
+
+        $ export DATABASE_USER=user
+        $ export DATABASE_PASSWORD=secret
+
+.. note::
+
+    You can also define the default value of any existing parameters using
+    special environement variables named after their corresponding parameter
+    prefixed with ``SYMFONY__`` after replacing dots by double underscores
+    (e.g. ``SYMFONY__FOO__BAR`` to set the default value of the ``foo.bar``
```

Comment:
let's keep using a real example here: `(e.g. `SYMFONY__DATABASE__USER` to set the default value of the `database.user` parameter).